subsys/net/openthread/rpc/client/ot_rpc_diag_client.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ot_rpc_ids.h>
+
+#include <nrf_rpc_cbor.h>
+
+#include <openthread/error.h>
+#include <openthread/dataset.h>
+#include <openthread/instance.h>
+#include <openthread/link.h>
+#include <openthread/thread.h>
+#include <sys/_stdint.h>
+
+NRF_RPC_GROUP_DECLARE(ot_group);
+
+static char version[256]; //to validate
+static otExtAddress mac = {.m8 = {0xF4, 0xCE, 0x36, 0x24, 0x95, 0x6A, 0xD1, 0xD0}};
+static otExtendedPanId ext_pan_id;
+static char thread_network_name[256];
+static otMeshLocalPrefix mesh_prefix;
+
+static void get_uint_t(ot_rpc_id id, void *result, size_t result_size) {
+	const size_t cbor_buffer_size = 0;
+	struct nrf_rpc_cbor_ctx ctx;
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, cbor_buffer_size);
+
+	nrf_rpc_cbor_cmd_rsp(&ot_group, id, &ctx);
+
+    zcbor_uint_decode(ctx.zs, result, result_size);
+
+	nrf_rpc_cbor_decoding_done(&ot_group, &ctx);
+}
+
+static struct zcbor_string get_string(ot_rpc_id id){
+	const size_t cbor_buffer_size = 0;
+    struct zcbor_string zst;
+	struct nrf_rpc_cbor_ctx ctx;
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, cbor_buffer_size);
+
+	nrf_rpc_cbor_cmd_rsp(&ot_group, id, &ctx);
+
+    zcbor_bstr_decode(ctx.zs, &zst);
+	nrf_rpc_cbor_decoding_done(&ot_group, &ctx);
+	return zst;
+}
+
+const char* otGetVersionString(void) {
+    struct zcbor_string zst = get_string(OT_RPC_CMD_GET_VERSION_STRING);
+
+	memcpy(version, zst.value, sizeof(version) < zst.len ? sizeof(version) : zst.len);
+
+	return version;
+}
+
+otError otDatasetGetActive(otInstance *aInstance, otOperationalDataset *aDataset) {
+	return 0;
+}
+
+uint8_t otLinkGetChannel(otInstance *aInstance) {
+	uint8_t ret;
+	get_uint_t(OT_RPC_CMD_LINK_GET_CHANNEL, &ret, sizeof(ret));
+	return ret;
+}
+
+const otMacCounters *otLinkGetCounters(otInstance *aInstance) {
+	return 0;
+}
+
+const otExtAddress *otLinkGetExtendedAddress(otInstance *aInstance) {
+	struct zcbor_string zst = get_string(OT_RPC_CMD_IF_EXTADDR);
+
+	memcpy(mac.m8, zst.value, sizeof(mac) < zst.len ? sizeof(mac) : zst.len);
```

Comment:
I think this should validate the length and report an error instead of dropping the superfluous bytes.